code/datums/components/ducttape.dm

```diff
@@ -47,23 +47,30 @@
 		return
 	if(!isturf(target))
 		return
-	if(!user.unEquip(I))
-		return
 	var/turf/source_turf = get_turf(I)
 	var/turf/target_turf = target
-	var/list/clickparams = params2list(params)
-	var/x_offset = text2num(clickparams["icon-x"]) - 16
-	var/y_offset = text2num(clickparams["icon-y"]) - 16
+	var/x_offset
+	var/y_offset
 	if(target_turf != get_turf(I)) //Trying to stick it on a wall, don't move it to the actual wall or you can move the item through it. Instead set the pixels as appropriate
 		var/target_direction = get_dir(source_turf, target_turf)//The direction we clicked
+		// Snowflake diagonal handling
+		if(target_direction == NORTHEAST || target_direction == SOUTHEAST || target_direction == NORTHWEST || target_direction == SOUTHWEST)
```

Comment:
I think you can make use of this: `GLOBAL_LIST_INIT(diagonals, list(NORTHEAST, NORTHWEST, SOUTHEAST, SOUTHWEST))` And do something like:

```js
if(target_direction in GLOB.diagonals)
```